### PR TITLE
Update directory expiration date after directory edit

### PR DESCRIPTION
**Pull request by rob-hills**
Update directory.expiration_dt after editing a directory to realign to any changes to start date or duration

This addresses issue #1311

### CODE REVIEW

File: tendenci/apps/directories/views.py [hunk 1]
```diff
@@ -314,6 +314,10 @@ def edit(request, id, form_class=DirectoryForm, template_name="directories/edit.
                 except IOError:
                     pass
                     #directory.logo = None
+            
+            # set the expiration date in case activation date and/or requested_duration have changed
+            directory.expiration_dt = directory.activation_dt + timedelta(days=directory.requested_duration)
```
Comment (jennyq) [inline tendenci/apps/directories/views.py:319]: Please add an if statement to make sure "directory.activation_dt" is not None. Also, this should only be available to superuser as regular users will not be able to edit the requested duration.

Comment (jennyq) [reply tendenci/apps/directories/views.py:319]: Also, please check if "directory.pricing" exists or not.

Comment (rob-hills) [reply tendenci/apps/directories/views.py]: @jennyq I've put in checks for directory.activation_dt AND for directory.requested_duration.

I had a look at the directory.add method and there are way more business rules in that around pricing, require_payment etc than there are in the edit method.  It occurs to me that maybe those rules should be pulled out into a single "gatekeeper" method that is called by both the add() and edit() methods?

**Comment by jennyq** [comment]
Thanks @rob-hills !

**Pull request merged**